querycheck/queryfilter/filter_by_display_name.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package queryfilter
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+)
+
+type filterByDisplayName struct {
+	displayNameCheck knownvalue.StringCheck
```

Comment:
nit: Similar comment to above about using `knownvalue.Check`, doesn't feel like too much of a stretch